Public/Src/Utilities/Native/IO/Helpers.cs

```diff
@@ -83,17 +78,10 @@ public static class Helpers
                 }
                 catch (Exception e)
                 {
-                    // Give the caller opportunity to re-throw the exception to break the retry loop
-                    if (rethrowExceptions)
-                    {
-                        throw;
-                    }
-                    
-                    // Only log exceptions if they are not being re-thrown already
-                    if (logExceptions)
-                    {
-                        Tracing.Logger.Log.RetryOnFailureException(Events.StaticContext, e.ToString());
-                    }
+                    onException?.Invoke(e);
+
+                    // For diagnostic purpose.
+                    Tracing.Logger.Log.RetryOnFailureException(Events.StaticContext, e.ToString());

```

Comment:
Now that it's up the caller to decide if/what to log, should we even bother calling this?

Reply:
For diagnostic purpose.